src/neptune_scale/api/attribute.py

```diff
@@ -211,25 +191,41 @@ def cleanup_path(path: str) -> str:
     Traceback (most recent call last):
     ...
     ValueError: Invalid path: `a//b/c`. Path components must not be empty.
+    >>> cleanup_path('a/ /b/c')
+    Traceback (most recent call last):
+    ...
+    ValueError: Invalid path: `a/ /b/c`. Path components must not be empty.
+    >>> cleanup_path('a/b/c ')
+    Traceback (most recent call last):
+    ...
+    ValueError: Invalid path: `a/b/c `. Path cannot contain leading or trailing whitespace.
     """
 
-    path = path.strip()
-    if path in ("", "/"):
+    if path.strip() in ("", "/"):
         raise ValueError(f"Invalid path: `{path}`.")
 
     if path.startswith("/"):
         path = path[1:]
 
     if path.endswith("/"):
         raise ValueError(f"Invalid path: `{path}`. Path must not end with a slash.")
-    if not all(path.split("/")):
+
+    if not all(x.strip() for x in path.split("/")):
         raise ValueError(f"Invalid path: `{path}`. Path components must not be empty.")
 
+    if path[0].lstrip() != path[0] or path[-1].rstrip() != path[-1]:
```

Comment:
hmm, path[0] and path[-1] are a substring of size 0 or 1, so there's no difference between lstrip() and rstrip()?

Reply:
Yeah I think I missed a `split()` or two